scheduler/scheduler.go

```diff
@@ -57,9 +60,19 @@ func NewTaskSchedule(taskChannelBuffer, concurrentWorkerLimit int) (*TaskSchedul
 }
 
 // Register adds a task or process to the TaskSchedule for execution as per its scheduling configuration.
-func (t *TaskSchedule) Register(task taskHandler) {
+func (t *TaskSchedule) Register(task Task, caps ...HighOrderTaskHandler) {
```

Comment:
Consider renaming the high order handler to:
```suggestion
func (t *TaskSchedule) Register(task Task, opts ...Options)
```

the `...options` is a common pattern in go, check this:
https://medium.com/@omid.jn/the-option-pattern-in-go-9edf55e88320

Reply:
Fixed.